server/midddleware/metrics.go

```diff
@@ -62,10 +62,10 @@ func countSpecificErrorMessage(fullMethod string, methodType, errSource string,
 func UnaryMetricsServerInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		methodType := "unary"
+		resp, err := handler(ctx, req)
 		tags := metrics.GetPreinitializedTagsFromFullMethod(info.FullMethod, methodType)
 		defer metrics.RequestsRespTime.Tagged(tags).Histogram("histogram", tally.DefaultBuckets).Start().Stop()
```

Comment:
This should be called before `resp, err := handler(ctx, req)` otherwise we are missing call latency.

Reply:
Right, this is fixed.